src/markov/markov_approx.jl

```diff
@@ -219,3 +223,543 @@ function estimate_mc_discrete{T}(X::Vector{T})
+                                       the innovations
+    - `discrete_var` only accepts non-singular variance-covariance matrices, `Psi`.
+- `Nm::Integer > 3` : Desired number of discrete points in each dimension
+
+##### Optional
+
+- `n_moments::Integer` : Desired number of moments to match. The default is 2.
+- `method::VAREstimationMethod` : Specify the method used to determine the grid
+                                  points. Accepted inputs are `Even()`.
+                                  Please see the paper for more details.
+                                  NOTE: `Quantile()` and `Quadrature()` are
+                                        not supported now.
+- `n_sigmas::Real` : If the `Even()` option is specified, `n_sigmas` is used to
+                     determine the number of unconditional standard deviations
+                     used to set the endpoints of the grid. The default is
+                     `sqrt(Nm-1)`.
+
+##### Returns
+
+- `P` : `Nm^M x Nm^M` probability transition matrix. Each row
+        corresponds to a discrete conditional probability
+        distribution over the state M-tuples in `X`
+- `X` : `M x Nm^M` matrix of states. Each column corresponds to an
+        M-tuple of values which correspond to the state associated
+        with each row of `P`
+
+##### NOTES
+
+- discrete_var only constructs tensor product grids where each dimension
+  contains the same number of points. For this reason it is recommended
+  that this code not be used for problems of more than about 4 or 5
+  dimensions due to curse of dimensionality issues.
+- Future updates will allow for singular variance-covariance matrices and
+  sparse grid specifications.
+
+"""
+function discrete_var(b::Union{Real, AbstractVector},
+                      B::Union{Real, AbstractMatrix},
+                      Psi::Union{Real, AbstractMatrix},
+                      Nm::Integer,
+                      n_moments::Integer=2,
+                      method::VAREstimationMethod=Even(),
+                      n_sigmas::Real=sqrt(Nm-1))
+
+    M, M_ = size(B, 1), size(B, 2)
+
+    # Check size restrictions on matrices
+    M == M_ || throw(ArgumentError("B must be a scalar or square matrix"))
+    size(b, 2) == 1 || throw(ArgumentError("b must be a column vector"))
+    M == length(b) || throw(ArgumentError("b must have the same number of rows as B"))
+
+    #% Check that Psi is a valid covariance matrix
+    isposdef(Psi) || throw(ArgumentError("Psi must be a positive definite matrix"))
+
+    # Check that Nm is a valid number of grid points
+    Nm >= 3 || throw(ArgumentError("Nm must be a positive interger greater than 3"))
+
+    # Check that n_moments is a valid number
+    if n_moments < 1 || !(n_moments % 2 == 0 || n_moments == 1)
+        error("n_moments must be either 1 or a positive even integer")
+    end
+
+    # Compute polynomial moments of standard normal distribution
+    gaussianMoment = zeros(n_moments)
```

Comment:
I know these are internal variable names, but would you please change them to have underscores instead of camel case?

In this case I'm asking for the variable to be `gaussian_moment`.

There are a few others throughout the code in this PR, but I'll only comment once to avoid excessive emails/comments.

In general the rule of thumb is to use `UpperCamelCase` for type names (including abstract, struct, and mutable struct) and use` lowercase_with_underscores` for variable and function names. I realize I wasn't ever clear about that before -- sorry.

Thanks!